app/index.js

```diff
@@ -147,11 +182,130 @@ module.exports = class extends yeoman {
         userProps.package = this.package;
 
         for (var i = 0; i < this.files.length; i++) {
-            this.fs.copyTpl(
-                this.templatePath(this.files[i]),
-                this.destinationPath(this.files[i].replace(/src\/main\/java/g, path.join(src, packageFolder))),
-                { userProps: userProps }
-            );
+            var skipFile = false;
+            if (showWsdl2Rest) {
+                skipFile = (testFileForWsdl2RestSkip(this.files[i]) == true);
+            } else {
+                if(this.files[i].localeCompare('pom.xml.wsdl2rest') == 0) {
+                    skipFile = true;
+                }
+            }
+
+            if (!skipFile) {
+                if(this.files[i].localeCompare('pom.xml.wsdl2rest') == 0) {
+                    var tempOutFile = "pom.xml";
+                    this.fs.copyTpl(
+                        this.templatePath(this.files[i]),
+                        this.destinationPath(tempOutFile.replace(/src\/main\/java/g, path.join(src, packageFolder))),
+                        { userProps: userProps }
+                    );
+                } else {
+                    this.fs.copyTpl(
+                        this.templatePath(this.files[i]),
+                        this.destinationPath(this.files[i].replace(/src\/main\/java/g, path.join(src, packageFolder))),
+                        { userProps: userProps }
+                    );
+                }
+            }
+        }
+
+        if (showWsdl2Rest) {
+            return wsdl2restGenerate(this.wsdl, this.outdirectory, this.camelDSL, showDebug);
         }
     }
 };
+
+var skipListForWsdl2RestFiles = [
+    'src/main/resources/META-INF/spring/camel-context.xml',
+    'src/main/resources/OSGI-INF/blueprint/blueprint.xml',
+    'pom.xml'
+];
+
+function testFileForWsdl2RestSkip(file) {
+    for(var i=0; i<skipListForWsdl2RestFiles.length; i++) {
+        if (skipListForWsdl2RestFiles[i] == file) return true;
+    }
+}
+
+function wsdl2restGenerate(wsdlUrl, outputDirectory, dsl, isDebug) {
+    if (dsl.indexOf('java') > 0) {
+        console.log(`Generating Rest DSL from SOAP for a Camel Java DSL is currently unsupported.`);
```

Comment:
I was able to generate the REST DSl choosing java DSl:

```
yo camel-project --wsdl2rest
     _                             _
    / \     _ __     __ _    ___  | |__     ___
   / _ \   | '_ \   / _` |  / __| | '_ \   / _ \
  / ___ \  | |_) | | (_| | | (__  | | | | |  __/
 /_/   \_\ | .__/   \__,_|  \___| |_| |_|  \___|
           |_|
       ____                              _
     /  ___|   __ _   _ __ ___     ___  | |
    |  |      / _` | | '_ ` _ \   / _ \ | |
    |  |___  | (_| | | | | | | | |  __/ | |
     \____|   \__,_| |_| |_| |_|  \___| |_|
 -----------------------------------------------
            Camel Project Generator
                 Version: 0.1.2
 -----------------------------------------------

? Your Camel project name (testCommandLine)
? Your Camel project name testCommandLine
? Your Camel version (2.22.2)
? Your Camel version 2.22.2
? Camel DSL type (blueprint, spring, or java) (java)
? Camel DSL type (blueprint, spring, or java) java
? Package name:  (com.testCommandLine)
? Package name:  koko.pl
? URL to the input WSDL (C:\Users\Aurelien Pupier\Desktop\Address.wsdl)
? URL to the input WSDL C:\Users\Aurelien Pupier\Desktop\Address.wsdl
? Name of the output directory for generated artifacts (src/main/java)
? Name of the output directory for generated artifacts src/main/java
Creating folders
Copying files
Calling wsdl2rest
   create CXF artifacts for specified WSDL at src/main/java
   create src/main/resources/META-INF/spring/camel-context.xml
   create README.md
   create src\main\java\koko\pl\routes\CamelRoute.java
   create src\main\java\koko\pl\routes\Launcher.java
```

Reply:
Interesting. You should have seen an error -- I'll check it out.

Reply:
Seems my validation method disappeared during one of my redo's - I'll work on getting that back in.